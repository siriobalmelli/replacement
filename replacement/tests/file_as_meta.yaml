---
# insert contents of file as metadata
# tests/file_as_meta.yaml
replacement:
  - text: dict
    input:
      # produces {"hello": "hi"}
      - dict: text
        key: hello
        input: |
          hi
      - dict: file
        key: contents
        input: hello.out

      - meta: file
        key: data
        input: prep.out
      - dict: dict
        prep: substitute
        input:
          also: $data

      - dict: file
        key: largefile
        input: recurse.out
...
