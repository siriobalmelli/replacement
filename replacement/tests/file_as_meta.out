---
hello: hi
contents: hello world
also: hello world
largefile: "recursed inline\nv1.1 tag \"my_awesome_tag\"\nmessage hello world\nhi\
  \ 5\nthis value may not exist - I exist I promise!"
...
